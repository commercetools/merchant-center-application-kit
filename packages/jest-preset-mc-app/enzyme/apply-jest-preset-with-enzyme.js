const path = require('path');

const resolveRelativePath = (relativePath) =>
  path.resolve(__dirname, relativePath);

const applyJestPresetWithEnzyme = ({ enzymeAdapterVersion, jestPreset }) => {
  switch (enzymeAdapterVersion) {
    case 16:
    case 17:
      break;
    default:
      throw new Error(
        `Invalid Enzyme adapter version "${enzymeAdapterVersion}". Supported versions are "16", "17".`
      );
  }
  return {
    ...jestPreset,
    // Custom preset config specific for Enzyme.
    snapshotSerializers: [
      ...(jestPreset.snapshotSerializers || []),
      'enzyme-to-json/serializer',
    ],
    setupFilesAfterEnv: [
      ...(jestPreset.setupFilesAfterEnv || []),
      resolveRelativePath(
        `./setup-test-framework-for-enzyme-${enzymeAdapterVersion}.js`
      ),
    ],
  };
};

module.exports = applyJestPresetWithEnzyme;
