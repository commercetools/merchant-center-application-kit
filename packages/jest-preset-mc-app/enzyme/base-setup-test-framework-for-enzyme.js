require('jest-enzyme');

const configureEnzymeExtensions = require('@commercetools/enzyme-extensions');
const commerceToolsEnzymeMatchers = require('@commercetools/jest-enzyme-matchers');
const ShallowWrapper = require('enzyme/ShallowWrapper');

expect.extend({
  toBeComponentWithName(received, actual) {
    const pass = received && received.displayName === actual;
    const message = () =>
      `expected ${received} ${pass ? 'not ' : ''} to be ${actual}`;
    return { message, pass };
  },
});

expect.extend(commerceToolsEnzymeMatchers);
configureEnzymeExtensions(ShallowWrapper);
