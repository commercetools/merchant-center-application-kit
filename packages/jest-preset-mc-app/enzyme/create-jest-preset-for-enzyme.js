const path = require('path');
const defaultPreset = require('../jest-preset');

const resolveRelativePath = (relativePath) =>
  path.resolve(__dirname, relativePath);

const createJestPresetForEnzyme = (version) => ({
  ...defaultPreset,
  snapshotSerializers: ['enzyme-to-json/serializer'],
  setupFilesAfterEnv: [
    resolveRelativePath('../setup-test-framework.js'),
    resolveRelativePath(`./setup-test-framework-for-enzyme-${version}.js`),
  ],
});

module.exports = createJestPresetForEnzyme;
