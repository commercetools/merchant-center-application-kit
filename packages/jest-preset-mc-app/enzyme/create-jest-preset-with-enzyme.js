const path = require('path');

const resolveRelativePath = (relativePath) =>
  path.resolve(__dirname, relativePath);

const createJestPresetForEnzyme = ({ enzymeAdapterVersion, jestPreset }) => {
  switch (enzymeAdapterVersion) {
    case 16:
    case 17:
      break;
    default:
      throw new Error(
        `Invalid Enzyme adapter version "${enzymeAdapterVersion}". Supported versions are "16", "17".`
      );
  }
  return {
    // Use default preset
    ...jestPreset,
    snapshotSerializers: [
      ...(jestPreset.snapshotSerializers || []),
      'enzyme-to-json/serializer',
    ],
    setupFilesAfterEnv: [
      ...(jestPreset.setupFilesAfterEnv || []),
      resolveRelativePath(
        `./setup-test-framework-for-enzyme-${enzymeAdapterVersion}.js`
      ),
    ],
  };
};

module.exports = createJestPresetForEnzyme;
