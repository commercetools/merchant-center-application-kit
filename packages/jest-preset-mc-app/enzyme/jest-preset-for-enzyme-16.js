const createJestPresetForEnzyme = require('./create-jest-preset-for-enzyme');

module.exports = createJestPresetForEnzyme(17);
