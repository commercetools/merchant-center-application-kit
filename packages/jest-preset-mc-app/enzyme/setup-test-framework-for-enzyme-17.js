require('./base-setup-test-framework-for-enzyme');
const Adapter = require('@wojtekmaj/enzyme-adapter-react-17');
const Enzyme = require('enzyme');

Enzyme.configure({ adapter: new Adapter(), disableLifecycleMethods: true });
