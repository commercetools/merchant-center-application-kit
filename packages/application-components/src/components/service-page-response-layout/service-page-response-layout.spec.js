import React from 'react';
import { shallow } from 'enzyme';
import ServicePageResponseLayout from './service-page-response-layout';

const createTestProps = props => ({
  imageSrc: '/assets/foo.svg',
  title: <div>{'title'}</div>,
  paragraph1: <div>{'title 1'}</div>,
  ...props,
});

describe('rendering', () => {
  describe('with only 1 paragraph', () => {
    let wrapper;
    let props;
    beforeEach(() => {
      props = createTestProps();
      wrapper = shallow(<ServicePageResponseLayout {...props} />);
    });
    it('outputs correct tree', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
  describe('with both paragraphs', () => {
    let wrapper;
    let props;
    beforeEach(() => {
      props = createTestProps({
        paragraph2: <div>{'title 2'}</div>,
      });
      wrapper = shallow(<ServicePageResponseLayout {...props} />);
    });
    it('outputs correct tree', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
  describe('with both paragraphs and a body content in between', () => {
    let wrapper;
    let props;
    beforeEach(() => {
      props = createTestProps({
        paragraph2: <div>{'title 2'}</div>,
        bodyContent: <div>{'content'}</div>,
      });
      wrapper = shallow(<ServicePageResponseLayout {...props} />);
    });
    it('outputs correct tree', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
  describe('with only one paragraph and the body content', () => {
    let wrapper;
    let props;
    beforeEach(() => {
      props = createTestProps({
        bodyContent: <div>{'content'}</div>,
      });
      wrapper = shallow(<ServicePageResponseLayout {...props} />);
    });
    it('outputs correct tree', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
