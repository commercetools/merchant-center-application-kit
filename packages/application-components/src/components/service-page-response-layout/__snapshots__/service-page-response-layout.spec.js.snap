// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering with both paragraphs and a body content in between outputs correct tree 1`] = `
<div
  className="container"
>
  <div
    className="content"
  >
    <div>
      <img
        className="graphic"
        src="/assets/foo.svg"
      />
    </div>
    <TextHeadline
      elementType="h2"
    >
      <div>
        title
      </div>
    </TextHeadline>
    <TextBody>
      <div>
        title 1
      </div>
    </TextBody>
    <div
      className="body"
    >
      <div>
        content
      </div>
    </div>
    <TextBody>
      <div>
        title 2
      </div>
    </TextBody>
  </div>
</div>
`;

exports[`rendering with both paragraphs outputs correct tree 1`] = `
<div
  className="container"
>
  <div
    className="content"
  >
    <div>
      <img
        className="graphic"
        src="/assets/foo.svg"
      />
    </div>
    <TextHeadline
      elementType="h2"
    >
      <div>
        title
      </div>
    </TextHeadline>
    <TextBody>
      <div>
        title 1
      </div>
    </TextBody>
    <TextBody>
      <div>
        title 2
      </div>
    </TextBody>
  </div>
</div>
`;

exports[`rendering with only 1 paragraph outputs correct tree 1`] = `
<div
  className="container"
>
  <div
    className="content"
  >
    <div>
      <img
        className="graphic"
        src="/assets/foo.svg"
      />
    </div>
    <TextHeadline
      elementType="h2"
    >
      <div>
        title
      </div>
    </TextHeadline>
    <TextBody>
      <div>
        title 1
      </div>
    </TextBody>
  </div>
</div>
`;

exports[`rendering with only one paragraph and the body content outputs correct tree 1`] = `
<div
  className="container"
>
  <div
    className="content"
  >
    <div>
      <img
        className="graphic"
        src="/assets/foo.svg"
      />
    </div>
    <TextHeadline
      elementType="h2"
    >
      <div>
        title
      </div>
    </TextHeadline>
    <TextBody>
      <div>
        title 1
      </div>
    </TextBody>
    <div
      className="body"
    >
      <div>
        content
      </div>
    </div>
  </div>
</div>
`;
